org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -925,25 +933,79 @@ public static void applyWorkspaceEdit(WorkspaceEdit wsEdit, String label) {
 		String name = label == null ? DEFAULT_LABEL : label;
 		Map<URI, Range> changedURIs = new HashMap<>();
 		CompositeChange change = toCompositeChange(wsEdit, name, changedURIs);
-		final var changeOperation = new PerformChangeOperation(change);
-		changeOperation.setUndoManager(RefactoringCore.getUndoManager(), name);
-		try {
-			ResourcesPlugin.getWorkspace().run(changeOperation, new NullProgressMonitor());
-
-			// Open the resource in editor if there is the only one URI
-			if (changedURIs.size() == 1) {
-				changedURIs.keySet().stream().findFirst().ifPresent(uri -> {
-					// Select the only start position of the range or the document start
-					Range range = changedURIs.get(uri);
-					Position start = range.getStart() != null ? range.getStart() : new Position(0, 0);
-					if (Display.getCurrent() != null) {
-						open(uri.toString(), new Range(start, start));
-					} else {
-						UI.getDisplay().asyncExec(() -> open(uri.toString(), new Range(start, start)));
-					}
-				});
+
+		if (wsEdit.getChangeAnnotations() != null && wsEdit.getChangeAnnotations().values().stream().anyMatch(ca -> ca.getNeedsConfirmation())) {
+			Refactoring refactoring = new Refactoring() {
+
+				@Override
+				public String getName() {
+					return label;
+				}
+
+				@Override
+				public RefactoringStatus checkInitialConditions(IProgressMonitor pm)
+						throws CoreException, OperationCanceledException {
+					return RefactoringStatus.create(Status.OK_STATUS);
+				}
+
+				@Override
+				public RefactoringStatus checkFinalConditions(IProgressMonitor pm)
+						throws CoreException, OperationCanceledException {
+					return RefactoringStatus.create(Status.OK_STATUS);
+				}
+
+				@Override
+				public Change createChange(IProgressMonitor pm) throws CoreException, OperationCanceledException {
+					return change;
+				}
+
+			};
+			RefactoringWizard wizard = new RefactoringWizard(refactoring,
+					RefactoringWizard.DIALOG_BASED_USER_INTERFACE |
+					RefactoringWizard.NO_BACK_BUTTON_ON_STATUS_DIALOG
+			) {
+
+				@Override
+				protected void addUserInputPages() {
+					//no inputs required
+				}
+
+			};
+			if (Display.getCurrent() == null) {
```

Comment:
for consistency within this method, could we invert the condition on `Display.getCurrent()` in the new or the old code so that they use the same condition and the the if/else is in the same order?

Reply:
I'll create runOnUiThread(Runnable) in the UI class and use it here in both places